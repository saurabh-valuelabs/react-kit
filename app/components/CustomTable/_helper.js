/*
 *
 * CustomTable helper
 *
 */

export function defaultFunction(text) {
  return text;
}
