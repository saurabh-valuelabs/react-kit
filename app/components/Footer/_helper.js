/*
 *
 * Navigation helper
 *
 */

import React from 'react';

import { Nav, NavItem, NavLink } from 'reactstrap';
import { Link } from 'react-router-dom';

export function returnItem(item) {
  return item.childrenList.map(innerItem => (
    <NavItem key={`footer-link-${innerItem.name}`} className="col-auto">
      <NavLink
        tag={Link}
        to={innerItem.udi ? innerItem.udi : innerItem.url}
        className="px-0"
      >
        {innerItem.name}
      </NavLink>
    </NavItem>
  ));
}

export function createNav(item) {
  return item.map(innerItem => (
    <Nav key={`footer-title-${item.headingText}`} className="row">
      {returnItem(innerItem)}
    </Nav>
  ));
}
