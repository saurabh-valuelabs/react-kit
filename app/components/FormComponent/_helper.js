/*
 *
 * FormComponent helper
 *
 */

export function defaultFunction(text) {
  return text;
}
