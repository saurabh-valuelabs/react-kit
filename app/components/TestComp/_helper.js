/*
 *
 * TestComp helper
 *
 */

export function defaultFunction(text) {
  return text;
}
