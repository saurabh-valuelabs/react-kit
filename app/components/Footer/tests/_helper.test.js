import { render } from 'react-testing-library';

import { createNav } from '../_helper';

import footerNav from '../../../config/_footerNav';

jest.mock('react-router-dom', () => ({ Link: 'a' }));

describe('Navigation helpers', () => {
  describe('Default Helper Function', () => {
    it('returns the expected value', () => {
      const {
        container: { firstChild },
      } = render(createNav(footerNav));
      expect(firstChild).toMatchSnapshot();
    });
  });
});
