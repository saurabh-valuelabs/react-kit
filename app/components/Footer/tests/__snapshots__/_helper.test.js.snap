// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation helpers Default Helper Function returns the expected value 1`] = `
<div
  class="col-12 col-md-4 col-lg-2"
>
  <h4>
    About Us
  </h4>
  <ul
    class="nav"
  >
    <li
      class="nav-item"
    >
      <a
        class="px-0 nav-link"
        to="/privacy-policy"
      >
        Privacy Policy
      </a>
    </li>
    <li
      class="nav-item"
    >
      <a
        class="px-0 nav-link"
        to="/list"
      >
        Term and Condition
      </a>
    </li>
  </ul>
</div>
`;
